lib/pingpong.c

```diff
@@ -358,7 +359,11 @@ CURLcode Curl_pp_readresp(curl_socket_t sockfd,
       pp->nread_resp += gotbytes;
       for(i = 0; i < gotbytes; ptr++, i++) {
         perline++;
-        if(*ptr == '\n') {
+        if(gotbytes >= 9 && strncmp(ptr,"* SEARCH ",9) == 0)
```

Comment:
Curl_pp_readresp() is a generic function used for FTP, IMAP, POP3 and SMTP so you can't add a non-conditional check for "SEARCH" like that! Also, I would encourage you to add a least one test case that verifies that your change works!